### PR TITLE
edge/devicetwin: fix index out of range in dealMetaDeviceOperation

### LINKED ISSUE #6999: edge/devicetwin: dealMetaDeviceOperation indexes resources[3] after validating len(resources) == 3
## What happened

In `edge/pkg/devicetwin/dtmanager/dmiworker.go`, `dealMetaDeviceOperation` validates the resource has exactly 3 segments:

```go
resources := strings.Split(message.Router.Resource, "/")
if len(resources) != 3 {                                  // dmiworker.go:114
    return fmt.Errorf("wrong resources %s", message.Router.Resource)
}
```

but the default branch of the resource-type switch then does:

```go
default:
    klog.Warningf("unsupported resource type %s", resources[3])   // dmiworker.go:195
```

`len(resources)` is guaranteed to be 3 here, so valid indices are `0–2`. `resources[3]` is out of range: any message whose middle segment is neither `device` nor `devicemodel` panics the `DMIWorker` goroutine and crashes `edgecore` instead of logging a warning.

## What you expected to happen

The unsupported type is logged (`resources[1]` is the type segment) and processing continues.

## How to reproduce it (as minimally and precisely as possible)

Send a `*model.Message` to the DMI worker with `Router.Resource = "ns/somethingelse/name"` and action `MetaDeviceOperation`. The callback panics with `index out of range [3] with length 3`.

## Anything else we need to know?

One-line fix: `resources[3]` → `resources[1]`. A unit test hitting the default branch would have caught this — the package's `dmiworker.go` has no test coverage for `dealMetaDeviceOperation`.

### PR DESCRIPTION
**Pull request by Priyanshubhartistm**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

`dealMetaDeviceOperation` in `edge/pkg/devicetwin/dtmanager/dmiworker.go` validates that `message.Router.Resource` splits into exactly 3`/`-separated segments (valid indices 0-2), but the `default` branch of
the subsequent resource-type switch logged `resources[3]`, which is always out of range at that point. Any DMI message whose middle resource segment was neither `device` nor `devicemodel` (e.g.
`"ns/somethingelse/name"`) panicked the DMIWorker goroutine, crashing edgecore instead of logging an "unsupported resource type" warning and continuing.

The fix changes `resources[3]` to `resources[1]` (the type segment), matching the code's evident intent. Also adds `dealMetaDeviceOperation` unit test coverage, including the previously-panicking default branch, since the package had none.

**Which issue(s) this PR fixes**:

Fixes #6999 

**Special notes for your reviewer**:

One-line fix; the bulk of the diff is the new test file. Verified the panic reproduces on `master` before the fix and is gone after, via the new `TestDealMetaDeviceOperation/UnsupportedResourceType` subtest.

**Does this PR introduce a user-facing change?**:

```release-note
Fix a panic in edgecore's DMIWorker when it receives a device metadata message with an unsupported resource type.
```

### CODE REVIEW

**Comment by DoisLONG** [comment]
Thanks for the fix.

The code change looks correct to me. After the resource string is validated to have exactly 3 segments, the resource type is resources[1], so using resources[3] in the default branch can panic.

The new test also covers the unsupported resource type path that would have hit this panic, and the previous test cleanup around err == nil has been addressed.

One thing to check: the current Main CI still has a failure in the edge integration test step. Could you take a look and confirm whether it is related to this PR, or rerun it if it is a flaky failure?

File: edge/pkg/devicetwin/dtmanager/dmiworker_test.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			err := dw.dealMetaDeviceOperation(nil, "", test.msg)
+			if (err != nil) != test.wantErr {
+				t.Errorf("dealMetaDeviceOperation() error = %v, wantErr %v", err, test.wantErr)
+				return
+			}
+			if test.errValue != "" && err.Error() != test.errValue {
+				t.Errorf("dealMetaDeviceOperation() error = %q, want %q", err.Error(), test.errValue)
+			}
```
[the remark this answers is not in the document]
Comment (Priyanshubhartistm) [reply edge/pkg/devicetwin/dtmanager/dmiworker_test.go]: done

**Comment by Priyanshubhartistm** [comment]
Thanks @DoisLONG for the review! I've updated everything, including rerunning the CI workflow. Please let me know if there's anything else that needs to be addressed.

**Review by DoisLONG — COMMENTED** [review]
Thanks for the update.

The DMIWorker fix itself looks correct to me. After the resource string is validated to have exactly 3 segments, `resources[1]` is the resource type, so using `resources[3]` in the default branch can panic.

The new `dealMetaDeviceOperation` tests also cover the unsupported-resource-type path and the input validation paths, and the previous nil-error handling concern in the test has been addressed.

One thing: this PR now also changes `cloud/pkg/csidriver/uds_test.go`, which looks unrelated to the DMIWorker panic fix. Could you remove that change from this PR, or explain why it is required here?

**Comment by Priyanshubhartistm** [comment]
Hi @DoisLONG, Thank you for the review! I've addressed the feedback:
Removed the unrelated cloud/pkg/csidriver/uds_test.go change 
The PR now contains only the dmiworker fix (dmiworker.go + dmiworker_test.go)

could you please take another look

**Review by DoisLONG — COMMENTED** [review]
Thanks for the update.

I checked the latest changes again. The unrelated `cloud/pkg/csidriver/uds_test.go` change has been removed, and the PR is now focused on the DMIWorker panic fix and the related unit test.

The fix looks correct to me: after the resource is validated to have exactly three segments, `resources[1]` is the resource type, so using it in the unsupported resource type branch is appropriate.

The added test also covers the previous panic path and related validation cases.

/lgtm

**Comment by Priyanshubhartistm** [comment]
Thank you for the review, @DoisLONG , @WillardHu
If everything looks good now, could you please /approve this PR when you have a chance? I'd really appreciate it.
